src/openapiGenerator/openapiGenerator.ts

```diff
@@ -0,0 +1,82 @@
+import { getSubcommand, SUBCOMMAND } from "./getSubcommand";
+import { GenerateOptions, GenerateDefaultOptions, GenerateOptionName } from "./options/generateOptions";
+import { OptionType, DefaultOptionType } from "./options/option.type";
+import { BatchOptions } from "./options/batchOptions";
+import { ListOptions } from "./options/listOptions";
+import { ConfigHelpOptions } from "./options/configHelpOptions";
+import { MetaOptions } from "./options/metaOptions";
+import { ValidateOptions } from "./options/validateOptions";
+import { CustomOptions, CustomOptionType, CustomOptionName } from "./options/customOptions";
+
+export function generatorFn() {
+    const { createCommand } = require("commander");
+    const program = createCommand();
+    [...GenerateOptions, ...BatchOptions, ...ListOptions, ...ConfigHelpOptions, ...MetaOptions, ...ValidateOptions, ...CustomOptions].map((item: OptionType) => {
+        program.option(item.option.join(", "), item.description, item.defaultValue);
+    });
+    program
+        .action(() => {
+            const { spawn } = require("child_process");
+            const args = process.argv.slice(2);
+            const subcommand: string = getSubcommand(args[0]);
+            args.shift();
+            let command: string = `node ./node_modules/@openapitools/openapi-generator-cli/bin/openapi-generator ${subcommand}`;
+            const extraOptions: Array<string> = formatExtraOption(args);
+            if (subcommand === SUBCOMMAND.GENERATE.toString()) {
```

Comment:
if you use my previous suggestion this will look much cleaner 
```
subcommand === "generate"
```

and generate will autocomplete and not accept any other value than the allowed ones

Reply:
changed